src/Filters/WhereDoesntHave.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace LaravelJsonApi\Eloquent\Filters;
+
+use LaravelJsonApi\Eloquent\Contracts\Filter;
+use LaravelJsonApi\Eloquent\Schema;
+
+class WhereDoesntHave implements Filter
+{
+    use Concerns\DeserializesValue;
+    use Concerns\IsSingular;
+
+    /**
+     * @var string
+     */
+    private string $relationName;
+
+    private Schema $schema;
+
+    /**
+     * SearchFilter constructor.
+     *
+     * @param string $relationName
+     * @param array $columns
+     */
+    public function __construct(string $relationName, Schema $schema)
```

Comment:
Same changes to this class. Order of constructor arguments should be:

1. $schema
2. $fieldName
3. $key = null

Reply:
Also make sure the classes' properties are in the same order and have docblocks.